### ISSUE TITLE
[Bug]: podman image scp fails with confusing error messages

### ISSUE THREAD

**Issue by Timmmm** [containers/podman#17090]
### Issue Description

If I run this:

```
podman image scp my_image:latest user@other_host::
```

It says this:

```
WARN[0000] Unknown connection name given. Please use system connection add to specify the default remote socket location 
Copying blob b12f34769c85 done  
Copying blob 7b3934b1b555 done  
Copying blob 9e0f46b6d29d done  
Copying blob 2e4e862242a9 done  
Copying blob ab61ffc4e8a6 done  
Copying blob 2cb8da93f979 done  
Copying blob 6517fb3ffeb2 done  
Copying config 96ae127d36 done  
Writing manifest to image destination
Storing signatures
Error: failed to connect: dial tcp: address other_host: missing port in address
```

Ok weird. Maybe it's some parsing bug to do with the `:`. I tried this:

```
podman image scp my_image:latest user@other_host
```

Seems to go a bit better:

```
Copying blob b12f34769c85 done  
Copying blob 7b3934b1b555 done  
Copying blob 9e0f46b6d29d done  
Copying blob 2e4e862242a9 done  
Copying blob ab61ffc4e8a6 done  
Copying blob 2cb8da93f979 done  
Copying blob 6517fb3ffeb2 done  
Copying config 96ae127d36 done  
Writing manifest to image destination
Storing signatures
[sudo] password for my_local_username: 
Getting image source signatures
Copying blob 2cb8da93f979 done  
Copying blob 7b3934b1b555 done  
Copying blob 2e4e862242a9 done  
Copying blob b12f34769c85 done  
Copying blob 9e0f46b6d29d done  
Copying blob ab61ffc4e8a6 done  
Copying blob 6517fb3ffeb2 done  
Copying config 96ae127d36 done  
Writing manifest to image destination
Storing signatures
Loaded image: localhost/verif_docker:latest
Error: exit status 125
```

Note that it randomly asked for the sudo password for my *local* account in the middle. First WTF, why does it need that? Second can it please *tell* me why it needs that?

Unfortunately it still failed with that `exit status 125` which is not a helpful message to put it mildly.

### Steps to reproduce the issue

Create a local docker image.

Use the above `podman image scp` commands.

It may be relevant that $HOME on the remote machine is NFS.

### Describe the results you received

It didn't work.

### Describe the results you expected

It should have worked.

### podman info output

```shell
Local:


host:
  arch: amd64
  buildahVersion: 1.27.1
  cgroupControllers: []
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: conmon-2.1.4-1.module+el8.7.0+17064+3b31f55c.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.4, commit: 64e1fe3ac604668d46b6efda338a9ba5a9b91b98'
  cpuUtilization:
    idlePercent: 90.04
    systemPercent: 1.42
    userPercent: 8.54
  cpus: 12
  distribution:
    distribution: '"rhel"'
    version: "8.7"
  eventLogger: file
  hostname: nb9JHHSQ3
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 529200513
      size: 1
    - container_id: 1
      host_id: 300000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 529202523
      size: 1
    - container_id: 1
      host_id: 300000
      size: 65536
  kernel: 6.0.11-1.el8.elrepo.x86_64
  linkmode: dynamic
  logDriver: k8s-file
  memFree: 1618714624
  memTotal: 16451919872
  networkBackend: cni
  ociRuntime:
    name: runc
    package: runc-1.1.4-1.module+el8.7.0+17064+3b31f55c.x86_64
    path: /usr/bin/runc
    version: |-
      runc version 1.1.4
      spec: 1.0.2-dev
      go: go1.18.7
      libseccomp: 2.5.2
  os: linux
  remoteSocket:
    path: /run/user/529202523/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_NET_RAW,CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-2.module+el8.7.0+17064+3b31f55c.x86_64
    version: |-
      slirp4netns version 1.2.0
      commit: 656041d45cfca7a4176f6b7eed9e4fe6c11e8383
      libslirp: 4.4.0
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.2
  swapFree: 0
  swapTotal: 0
  uptime: 20h 10m 11.00s (Approximately 0.83 days)
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.access.redhat.com
  - registry.redhat.io
  - docker.io
store:
  configFile: /home/me/.config/containers/storage.conf
  containerStore:
    number: 14
    paused: 0
    running: 0
    stopped: 14
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/me/.local/share/containers/storage
  graphRootAllocated: 362630144000
  graphRootUsed: 61168263168
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 12
  runRoot: /run/user/529202523/containers
  volumePath: /home/me/.local/share/containers/storage/volumes
version:
  APIVersion: 4.2.0
  Built: 1666812227
  BuiltTime: Wed Oct 26 20:23:47 2022
  GitCommit: ""
  GoVersion: go1.18.7
  Os: linux
  OsArch: linux/amd64
  Version: 4.2.0
```

Remote:

```
host:
  arch: amd64
  buildahVersion: 1.27.1
  cgroupControllers: []
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: conmon-2.1.4-1.module+el8.7.0+1078+e72fcd4f.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.4, commit: 6806ea86c9ad5ad2677388246a64dd8079924180'
  cpuUtilization:
    idlePercent: 99.59
    systemPercent: 0.07
    userPercent: 0.35
  cpus: 8
  distribution:
    distribution: '"rocky"'
    version: "8.7"
  eventLogger: file
  hostname: in-czba-psd0002.base.codasip.com
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 529200513
      size: 1
    uidmap:
    - container_id: 0
      host_id: 529201688
      size: 1
  kernel: 4.18.0-425.3.1.el8.x86_64
  linkmode: dynamic
  logDriver: k8s-file
  memFree: 23196172288
  memTotal: 33481314304
  networkBackend: cni
  ociRuntime:
    name: runc
    package: runc-1.1.4-1.module+el8.7.0+1078+e72fcd4f.x86_64
    path: /usr/bin/runc
    version: |-
      runc version 1.1.4
      spec: 1.0.2-dev
      go: go1.18.4
      libseccomp: 2.5.2
  os: linux
  remoteSocket:
    path: /run/user/529201688/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_NET_RAW,CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-2.module+el8.7.0+1078+e72fcd4f.x86_64
    version: |-
      slirp4netns version 1.2.0
      commit: 656041d45cfca7a4176f6b7eed9e4fe6c11e8383
      libslirp: 4.4.0
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.2
  swapFree: 2147479552
  swapTotal: 2147479552
  uptime: 19h 12m 12.00s (Approximately 0.79 days)
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.access.redhat.com
  - registry.redhat.io
  - docker.io
store:
  configFile: /home/user/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/user/.local/share/containers/storage
  graphRootAllocated: 10737483776
  graphRootUsed: 1800339456
  graphStatus:
    Backing Filesystem: nfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 0
  runRoot: /run/user/529201688/containers
  volumePath: /home/user/.local/share/containers/storage/volumes
version:
  APIVersion: 4.2.0
  Built: 1667932451
  BuiltTime: Tue Nov  8 19:34:11 2022
  GitCommit: ""
  GoVersion: go1.18.4
  Os: linux
  OsArch: linux/amd64
  Version: 4.2.0

```
```


### Podman in a container

No

### Privileged Or Rootless

Rootless

### Upstream Latest Release

No

### Additional environment details

_No response_

### Additional information

_No response_

**Comment by Timmmm** [comment]
Last bit of the logs with debugging on:

```
Storing signatures
DEBU[0007] setting image creation date to 2023-01-12 11:22:08.130979692 +0000 UTC 
DEBU[0007] reusing image ID "96ae127d365f0ac585c2c2eec57f463dffc7eb97f57ffdb9d935b95bb4213d25" 
DEBU[0007] saved image metadata "{\"signatures-sizes\":{\"sha256:8cb942b1da5a7ec714133f330e73b45f152467581900e4ef1c8fb7e636496d4e\":[]}}" 
DEBU[0007] added name "localhost/verif_docker:latest" to image "96ae127d365f0ac585c2c2eec57f463dffc7eb97f57ffdb9d935b95bb4213d25" 
DEBU[0007] Called load.PersistentPostRunE(/usr/bin/podman --log-level=debug load --input /var/tmp/podman2393914383) 
Loaded image: localhost/verif_docker:latest
Error: exit status 125
```

**Comment by braatvedt** [comment]
Just to add to this I was attempting a podman image scp from a new machine and was getting an unhelpful exit status 255. I should point out I'm using `--ssh=native` and using an ed25519 key, as rsa isn't supported.

After hours of searching and attempting things, it turns out you need a config file in .ssh so this sorted it out:

```
touch ~/.ssh/config
chmod 600 ~/.ssh/config
```

If that helps anyone else. Some clearer error messages would be great though.

**Comment by rhatdan** [comment]
@cdoern any comment on this?

**Comment by drewlsvern** [comment]
I'm experiencing the exact same issue as @Timmmm.

Any update on this?

I can provide more info to help try and debug this.

**Comment by drewlsvern** [comment]
@Timmmm I have been migrating from docker to podman, most of my workflow now revolves around podman now. 

I have workaround with Ansible, but it's much more complicated that would be necessary if this issue could be resolved.

**Comment by drewlsvern** [comment]
I decided to give this another try last week and managed to get it working. I re-read the docs and this [blog article](https://www.redhat.com/sysadmin/podman-transfer-container-images-without-registry) .

I setup the connection using `podman system connection add` and now it works flawlessly. I believe the first time I attempted this I wasn't setting the podman socket correctly.

From my point of view this issue can be closed.
